src/plugins/image_embeddable/public/components/image_embeddable.tsx

```diff
@@ -63,7 +63,9 @@ export const ImageEmbeddable = ({ api, filesClient }: ImageEmbeddableProps) => {
       }}
     >
       <ImageViewer
-        data-rendering-count={1} // TODO: Remove this as part of https://github.com/elastic/kibana/issues/179376
+        // TODO: Remove data-shared-item and data-rendering-count as part of https://github.com/elastic/kibana/issues/179376
+        data-shared-item={''}
```

Comment:
Does this mean that each embeddable will be responsible for setting data-shared-item and data-rendering-count?

Reply:
Temporarily yes. And yep this means it need to be documented and communicated well to the rest. Unless we want to bring back the initial solution that adds this attribute to `src/plugins/presentation_panel/public/panel_component/presentation_panel_internal.tsx` and focus on making the test pass. Either by looking more into the implementation and fixing it or even by changing the test (not sure if this is the right approach, but how the whole data-shared-item works looks mysterious)

I was looking into this file `test/functional/apps/dashboard/group5/data_shared_attributes.ts` line 59. There are some extra `undefined` items added in the `sharedData` array if I console log it. I can make the test pass by filtering out the undefined elements `expect(sharedData.map((item) => item.title).filter((item) => item !== undefined)).to.eql(originalPanelTitles)`

Let's further discuss about it tomorrow.

Reply:
I don't think changing the test is the right approach - these attributes are used for actual reporting and not **just** the reporting tests, so we should keep the tests the same if possible. It's definitely annoying that each embeddable currently has to handle their own `data-*` attributes, but unless we want to go into every legacy embeddable and remove their `data-*` logic, this currently seems like the best approach :)

This will all be cleaned up as part of https://github.com/elastic/kibana/issues/179376

Reply:
Yep makes total sense! Indeed this currently seems like the best approach, totally agree.